crates/hir/src/semantics/source_to_def.rs

```diff
@@ -499,7 +504,12 @@ impl SourceToDefCtx<'_, '_> {
         let parent = |this: &mut Self, node: InFile<&SyntaxNode>| match node.value.parent() {
             Some(parent) => Some(node.with_value(parent)),
             None => {
-                let macro_file = node.file_id.macro_file()?;
+                let macro_file = if let Some(macro_file) = node.file_id.macro_file() {
+                    macro_file
+                } else {
+                    let file_id = node.file_id.file_id()?;
+                    *this.cache.included_file_cache.get(&file_id)?
+                };
```

Comment:
If the current node is inside the `include!`d file, this recursive parent search was stopped on `macro_file()?` at the file's outer node.
But if this was `None` it was fallbacked to the root module. So, it worked fine if the file was `include!`d to root and caused the weird behaviour #18314 if it was inside a block module.